structures/occupancytypes.go

```diff
@@ -379,6 +413,56 @@ func res11snbPier() OccupancyTypeStochastic {
 	return OccupancyTypeStochastic{Name: "RES1-1SNB-PIER", StructureDFF: sdf, ContentDFF: cdf}
 }
 
+func res11snbPierMedwave() OccupancyTypeStochastic {
+
+	structurewavexs := []float64{-4.0, -3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0}
+	structurewaveys := []float64{2, 2, 3, 4, 14, 32, 42, 48, 56, 61, 68, 72, 77, 81, 84, 86, 89, 91, 94, 96, 97}
+	var structuredamagefunction = paireddata.PairedData{Xvals: structurewavexs, Yvals: structurewaveys}
+
+	contentwavexs := []float64{-4.0, -3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0}
+	contentwaveys := []float64{2, 2, 3, 4, 14, 32, 42, 48, 56, 61, 68, 72, 77, 81, 84, 86, 89, 91, 94, 96, 97}
+	var contentdamagefunction = paireddata.PairedData{Xvals: contentwavexs, Yvals: contentwaveys}
+
+	sm := make(map[hazards.Parameter]interface{})
+	var sdf = DamageFunctionFamilyStochastic{DamageFunctions: sm}
+
+	cm := make(map[hazards.Parameter]interface{})
+	var cdf = DamageFunctionFamilyStochastic{DamageFunctions: cm}
+	//Default hazard.
+	sdf.DamageFunctions[hazards.Default] = structuredamagefunction
+	cdf.DamageFunctions[hazards.Default] = contentdamagefunction
+	//Waves Hazard
+	sdf.DamageFunctions[hazards.WaveHeight] = structuredamagefunction
+	cdf.DamageFunctions[hazards.WaveHeight] = contentdamagefunction
+
+	return OccupancyTypeStochastic{Name: "RES1-1SNB-PIER_MEDWAVE", StructureDFF: sdf, ContentDFF: cdf}
+}
+
+func res11snbPierHighwave() OccupancyTypeStochastic {
+
+	structurewavexs := []float64{-4.0, -3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0}
+	structurewaveys := []float64{5, 8, 10, 12, 20, 38, 50, 58, 66, 73, 82, 86, 92, 97, 100, 100, 100, 100, 100, 100, 100}
+	var structuredamagefunction = paireddata.PairedData{Xvals: structurewavexs, Yvals: structurewaveys}
+
+	contentwavexs := []float64{-4.0, -3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0}
+	contentwaveys := []float64{5, 8, 10, 12, 20, 38, 50, 58, 66, 73, 82, 86, 92, 97, 100, 100, 100, 100, 100, 100, 100}
+	var contentdamagefunction = paireddata.PairedData{Xvals: contentwavexs, Yvals: contentwaveys}
+
+	sm := make(map[hazards.Parameter]interface{})
+	var sdf = DamageFunctionFamilyStochastic{DamageFunctions: sm}
+
+	cm := make(map[hazards.Parameter]interface{})
+	var cdf = DamageFunctionFamilyStochastic{DamageFunctions: cm}
+	//Default hazard.
+	sdf.DamageFunctions[hazards.Default] = structuredamagefunction

```

Comment:
i think this should probably be hazards.Wave|hazards.Salinity|hazards.Depth right?

Reply:
Oh, i see, sorrry, that comment should apply to line 460 and 461 and i think it has to be all three parameters

Reply:
I think I have a different understanding of what the wave curves represent. First, the wave functions may be applied to both saltwater and freshwater (the few locations of A zones around the Great Lakes. Second, whereas the saltwater inundation damage curve are both a function of the depth of inundation and the presence of salt in the water, the waves damage curves are a function solely of the effect of the wave action on the structure. I don't think that the wave functions represent the joint effect of waves and inundation on a structure. If we are interested in modeling the joint effect of waves and inundation (salt or fresh) we could do that following the methodology in beach-fx 
![image](https://user-images.githubusercontent.com/25163165/119040266-0c5c9c80-b96a-11eb-9158-cdf85956598a.png)

Reply:
oh no, i think we have the same understanding of what the curves represent, i think the misunderstanding is in how the map of curves work in the hazard function family. The index for the map is a bitflag. The bitflag must represent all elements of the hazard at play to get to the curve. 
so the wave based curves respond to a hazard that must have an indication of salinity, wave height, and depth to compute the right damage, so the lookup requires that the hazard has all 3 values. The XOR on bitwize operations works more like an and.